### PR TITLE
Adds generic search query support

### PR DESCRIPTION
**Pull request by hoddr**

### CODE REVIEW

File: voxel51/users/query.py [hunk 1]
```diff
@@ -115,6 +115,20 @@ def add_search_or(self, field, search_strs):
         '''
         return self.add_search(field, "|".join(search_strs))
 
+    def add_generic_search(self, search_str):
```
Comment (brimoor) [inline voxel51/users/query.py]: "generic" search is not very meaningful. Something can be generic in many ways. As a user scrolling through the docs, I am more likely to understand what this does if I see:

```py
    def add_search_any_field(self, search_str):
        '''Adds the given search that matches any field to the query.

        The resulting query will return results where any field matches the
        specified search string.

        ...
    ```
```

File: voxel51/users/query.py [hunk 2]
```diff
@@ -115,6 +115,20 @@ def add_search_or(self, field, search_strs):
...
+        '''Adds generic search, e.g. search across all searchable fields.
+
+        Overwritten if used in conjunction with other search functions.
```
Comment (brimoor) [inline voxel51/users/query.py]: Why is this the case? I haven't looked at the backend implementation, but I don't see why you couldn't support:
```
.add_search("status", "COMPLETE")
.add_search_any_field("xxxx")
```
to match completed jobs that contain "xxxx" somewhere. It's just another `WHERE <condition> AND <other condition>` clause.

**Review by brimoor — CHANGES_REQUESTED** [review]

File: voxel51/users/query.py (hunk 2 above)
Comment (hoddr) [reply voxel51/users/query.py]: Deprecated. Now possible.

**Review by brimoor — APPROVED** [review]
Thanks for making the change!

**Pull request merged**